judge-control-app/src/remote_exec/ssh.rs

```diff
@@ -0,0 +1,122 @@
+            .await
+            .map_err(|e| SshExecError::InternalServerError(e))?;
+        let output = self
+            .exec_inner_with_timeout(cmd, channel, execution_time_limit)
+            .await
+            .map_err(|e| SshExecError::RemoteServerError(e))?;
+        Ok(output)
+    }
+}
+
+impl<AddrType: ToSocketAddrs> SshConnection<AddrType> {
+    // SSH接続の確立
+    async fn connect_with_timeout(
+        &self,
+        connection_time_limit: Duration,
+    ) -> Result<Channel, InternalServerError> {
+        let connect_future = async move {
+            let tcp = TcpStream::connect(&self.addrs)
+                .await
+                .context("Failed to connect to the SSH server")?;
+            let mut sess = Session::new().context("Failed to create a new SSH session")?;
+            sess.set_tcp_stream(tcp);
+            sess.handshake()
+                .context("Failed to perform SSH handshake")?;
+            sess.userauth_password(&self.username, &self.password)
+                .context("Failed to authenticate with the SSH server")?;
+            let chan = sess
+                .channel_session()
+                .context("Failed to open a new channel for SSH")?;
+            Ok(chan)
+        };
+        let timeout_future = async move {
+            let result = timeout(connection_time_limit, connect_future)
+                .await
+                .map_err(|e| anyhow::Error::from(e))
+                .context("Connection time limit exceeded")?;
+            result
+        };
+        let result: Result<Channel, InternalServerError> = timeout_future
+            .await
+            .map_err(|e| InternalServerError(e).into());
+        result
+    }
+
+    // コマンドの実行
+    async fn exec_inner_with_timeout(
+        &self,
+        cmd: &str,
+        mut chan: Channel,
+        execution_time_limit: Duration,
+    ) -> Result<String, RemoteServerError> {
+        let exec_future = async move {
+            chan.exec(cmd)
+                .context("Failed to execute the command via SSH")?;
+            let mut output = String::new();
+            chan.read_to_string(&mut output)
+                .context("Failed to read the output from SSH")?;
+            Ok(output)
+        };
+        let timeout_future = async move {
+            let start_time = tokio::time::Instant::now();
+            let result = timeout(execution_time_limit + Duration::from_secs(1), exec_future)
+                .await
+                .map_err(|e| anyhow::Error::from(e))
+                .context("Execution time limit exceeded")?;
+            let elapsed = tokio::time::Instant::now().duration_since(start_time);
+            if elapsed >= execution_time_limit {
+                return Err(anyhow::anyhow!("Execution time limit exceeded"));
+            }
+            result
+        };
+        let result: Result<String, RemoteServerError> = timeout_future
+            .await
+            .map_err(|e| RemoteServerError(e).into());
```

Comment:
useless conversion to the same type
```suggestion
            .map_err(RemoteServerError);
```